realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -435,6 +435,9 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedRealm_nativeGetTable(JNIEnv
         auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
         if (!shared_realm->read_group().has_table(name)) {
             std::string name_str = name;
+            if (name_str.find(TABLE_PREFIX) == 0) {
```

Comment:
I totally think you are absolutely right.  Eventually the whole table prefix should disappear from Java.  If the core/os folks decide to change it to "foobar_", I don't even want to hear about it.  ...until then, though...